From 9d9c925385d75b95bf9b537120516b7cdd3e218b Mon Sep 17 00:00:00 2001
From: Bert Outtier <bert.outtier@spectricity.com>
Date: Wed, 7 Oct 2026 18:13:44 +0200
Subject: [PATCH] kbuild: include the dtb rules when building a single .dtbo

Building a single overlay in a directory without dtb-y, e.g.

  make rpi-7inch-dsi.dtbo

with the .dtso copied to arch/arm/boot/dts, fails with:

  make[3]: *** No rule to make target 'arch/arm/boot/dts/rpi-7inch-dsi.dtbo'.  Stop.

Makefile.dtbs, which has the %.dtbo rule, is only included when the
targets contain a .dtb, .dtb.o or .dtbo.o file. Also include it for a
.dtbo target, as is done for a single .dtb.

Upstream: not yet submitted
Signed-off-by: Bert Outtier <bert.outtier@spectricity.com>
---
 scripts/Makefile.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/Makefile.build b/scripts/Makefile.build
index 5037f4715..ba0fcb3ea 100644
--- a/scripts/Makefile.build
+++ b/scripts/Makefile.build
@@ -557,7 +557,7 @@ targets += $(filter-out $(single-subdir-goals), $(MAKECMDGOALS))
 targets := $(filter-out $(PHONY), $(targets))
 
 # Now that targets is fully known, include dtb rules if needed
-ifneq ($(need-dtbslist)$(dtb-y)$(dtb-)$(filter %.dtb %.dtb.o %.dtbo.o,$(targets)),)
+ifneq ($(need-dtbslist)$(dtb-y)$(dtb-)$(filter %.dtb %.dtbo %.dtb.o %.dtbo.o,$(targets)),)
 include $(srctree)/scripts/Makefile.dtbs
 endif
 
-- 
2.53.0

